src/Schema/Events/ImpressionTracking.ts

```diff
@@ -104,6 +104,28 @@ export interface ProgressiveOnboardingTooltipViewed {
   type: string
 }
 
+/**
+ * A user viewed a toast. `subject` will describe what toast it is.
+ *
+ * This schema describes events sent to Segment from [[viewedToast]]
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "viewedToast",
+ *    context_module: "artworkRecommendations",
+ *    context_screen_owner_type: "artworkGrid",
+ *    subject: "Set an artwork budget in your profile at any time."
+ *  }
+ * ```
+ */
+export interface ViewedToast {
+  action: ActionType.viewedToast
+  context_module: ContextModule
+  context_screen_owner_type: OwnerType
+  subject: string
```

Comment:
I wonder if `subject` is needed, I would have assumed we would be saving an `id` that doesn't change regardless of the copy

Reply:
I think subject is fine here – we use this pattern for other buttons / toasts where it's useful to have the copy context but we don't really need a consistent identifier across copy changes. I don't think we often use the `subject` for querying. If we ever _do_ need consistent query support – perhaps something like `analyticsID` would be an appropriate addition to these schema events – but I haven't heard a need/desire for that yet.